### DISCUSSION TITLE
Unable to use env variable in Github Actions if statement

### DISCUSSION THREAD

**Discussion by ravgeetdhillon** [community/community#24958] [Actions]
<p>I am trying to run a simple workflow like this.</p>
<pre><code class="lang-auto">name: Create Release

on:
  push:
    branches:
      - master

jobs:
  job1:
      - name: Set Environment
        run: |
          string="Hello World"
          echo "string=$string" &gt;&gt; $GITHUB_ENV

  job2:
    needs: job1
    if: contains(env.string, 'Hello')
    runs-on: ubuntu-latest
    steps:
      - name: Set up Repository
        uses: actions/checkout@v2
</code></pre>
<p>However, I get an error</p>
<pre><code class="lang-auto">The workflow is not valid. .github/workflows/create-release.yml Unrecognized named-value: 'env'
</code></pre>
<p>I am not able to access the env variable <strong>string</strong> in the if statement. How can solve this issue?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Environment variables aren’t persistent across jobs, you’ll need to use job outputs for that:</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idoutputs" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idoutputs" target="_blank" rel="noopener">Workflow syntax for GitHub Actions - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idoutputs" target="_blank" rel="noopener">//docs.github.com/en/actions/reference/workflow-syntax-for-github-actions</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
